From 0996ec3fc6ea3ec3c76392f9aa15a283060f95fa Mon Sep 17 00:00:00 2001
From: Sil Vilerino <sivileri@microsoft.com>
Date: Fri, 25 Aug 2023 13:05:54 -0400
Subject: [PATCH 607/834] frontends/va: Add profile param when querying
 PIPE_VIDEO_CAP_ENC_QUALITY_LEVEL

Pass the specific profile as drivers may only support
quality levels for specific codecs or profiles.

Reviewed-by: Ruijing Dong <ruijing.dong@amd.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25819>
---
 src/gallium/frontends/va/picture.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gallium/frontends/va/picture.c b/src/gallium/frontends/va/picture.c
index 0a4b5ffc8c3..ee845b3a42f 100644
--- a/src/gallium/frontends/va/picture.c
+++ b/src/gallium/frontends/va/picture.c
@@ -1141,7 +1141,7 @@ vlVaEndPicture(VADriverContextP ctx, VAContextID context_id)
          context->desc.h264enc.frame_num_cnt++;
 
       /* keep other path the same way */
-      if (!screen->get_video_param(screen, PIPE_VIDEO_PROFILE_UNKNOWN,
+      if (!screen->get_video_param(screen, ProfileToPipe(context->templat.profile),
                                   context->decoder->entrypoint,
                                   PIPE_VIDEO_CAP_ENC_QUALITY_LEVEL)) {
 
-- 
2.42.0

